### PR TITLE
Add setup for Haskell server

### LINKED ISSUE #38: Set up infra for Haskell server
Set up the infrastructure for the Haskell server (`applyCode`, fee estimation, etc...) in this repository

### PR DESCRIPTION
**Pull request by ngua**
Closes #38

This adds the necessary setup for the Haskell server we'll be relying on for the time being. The flake combines the outputs of the Haskell and Purescript projects, but keeps the `devShell` solely for the PS project (`hsDevShell` is for the Haskell project; I was thinking it didn't make a lot of sense to pull in a bunch of Haskell dependencies just to work on the Purescript project)

### CODE REVIEW

**Comment by ngua** [comment]
No idea what's happening with Hercules:

```
FatalError {fatalErrorMessage = "Worker failed with exit status: -6"}
```

**Comment by ngua** [comment]
No idea what's happening with Hercules:

```
FatalError {fatalErrorMessage = "Worker failed with exit status: -6"}
```

**Comment by ngua** [comment]
No idea what's happening with Hercules:

```
FatalError {fatalErrorMessage = "Worker failed with exit status: -6"}
```

**Comment by ngua** [comment]
Apparently Hercules has an issue with two flakes in the repo (there really should be only one in most circumstances, but I was hoping it would with two). Reverted to original combined flake now

File: server/cabal.project [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package cardano-wallet
+  optimization: False
+package cardano-wallet-core
+  optimization: False
+package cardano-wallet-cli
+  optimization: False
+package cardano-wallet-launcher
+  optimization: False
+package cardano-wallet-core-integration
+  optimization: False
```
Comment (adamczykm) [inline server/cabal.project:43]: Why are optimizations disabled?

File: server/cardano-browser-tx-server.cabal [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    TypeApplications
+    TypeFamilies
+    TypeOperators
+    TypeSynonymInstances
+
+executable cardano-browser-tx-server
+  import:           common-language
+  default-language: Haskell2010
+  ghc-options:
+    -Wall -Wcompat -Wincomplete-uni-patterns -Wredundant-constraints
```
Comment (adamczykm) [inline server/cardano-browser-tx-server.cabal:48]: And here as well?

**Review by adamczykm — COMMENTED** [review]
I have no deeper experience with Nix. I think someone more proficient should take a look as well.

File: server/cardano-browser-tx-server.cabal [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    FlexibleContexts
+    FlexibleInstances
+    GeneralizedNewtypeDeriving
+    ImportQualifiedPost
+    LambdaCase
+    MonoLocalBinds
+    MultiParamTypeClasses
+    NumericUnderscores
+    OverloadedStrings
+    OverloadedStrings
```
Comment (vvtran) [inline server/cardano-browser-tx-server.cabal:28]:
```
OverloadedStrings
OverloadedStrings
QuasiQuotes
QuasiQuotes
RecordWildCards
RecordWildCards
```
few duplicates here.

**Review by vvtran — CHANGES_REQUESTED** [review]
Not a Nix expert so not sure if I'm best qualified.

**Comment by vvtran** [comment]
Do we want to add `/server/dist-newstyle` and `/server/dist-newstyle/` to `.gitignore`?

File: server/cardano-browser-tx-server.cabal (hunk 3 above)
Comment (ngua) [reply server/cardano-browser-tx-server.cabal:28]: Good catch

File: server/cabal.project (hunk 1 above)
Comment (ngua) [reply server/cabal.project:43]: This is inherited from Plutus projects, IOG also configures their builds like this. They claim that it has little noticeable impact on performance but cuts build time by 50%

Comment (ngua) [reply server/cabal.project:43]: There was an explanatory comment there, not sure why I deleted it. I'll put it back

File: server/cardano-browser-tx-server.cabal (hunk 2 above)
Comment (ngua) [reply server/cardano-browser-tx-server.cabal:48]: Sorry, what do you mean?

**Comment by ngua** [comment]
@vvtran There in the .gitignore on my estimate-fees branch, but I'll add it here too

**Review by vvtran — APPROVED** [review]

**Pull request merged**